core/dbt/logger.py

```diff
@@ -344,8 +344,9 @@ def process(self, record):
 
 class ScrubSecrets(logbook.Processor):
     def process(self, record):
+        subbed = str(record.message)
         for secret in get_secret_env():
-            record.message = record.message.replace(secret, "*****")
+            record.message = subbed.replace(secret, "*****")
```

Comment:
definitely open to a smarter / more pythonic way to do this!

Reply:
Looks fine to me, but I'm no expert and phyonic-ness.

Reply:
FWIW the DBT_ENV_SECRET vars are prepended with `********`

Reply:
It's perfectly reasonable IMO.. If I was being terribly pedantic I might suggest skipping assigning the stringified value to a var like so:

```
for secret in get_secret_env():
            record.message = str(record.message).replace(secret, "*****")
            
```

Reply:
I agree with @iknox-fa and would like to skip the assignment. But also won't block merging on that.